packages/payment-widget/src/lib/components/payment-confirmation.svelte

```diff
@@ -0,0 +1,418 @@
+  import CopyIcon from "@requestnetwork/shared-icons/copy-icon.svelte";
+  import ExchangeIcon from "@requestnetwork/shared-icons/exchange.svelte";
+  import InfoCircleIcon from "@requestnetwork/shared-icons/info-circle.svelte";
+  import type { Web3Modal } from "@web3modal/ethers5";
+  import { onDestroy, onMount } from "svelte";
+  import type { Currency, PaymentStep } from "../types";
+  import { chains } from "../utils/chains";
+  import { NETWORK_LABEL } from "../utils/currencies";
+  import {
+    handleRequestPayment,
+    prepareRequestParameters,
+  } from "../utils/request";
+  import Spinner from "./spinner.svelte";
+  import WalletInfo from "./wallet-info.svelte";
+
+  export let selectedCurrency: Currency;
+  export let amountInUSD: number;
+  export let sellerAddress: string;
+  export let currentPaymentStep: PaymentStep;
+  export let web3Modal: Web3Modal | null;
+  export let isConnected: boolean;
+  export let persistRequest: boolean;
+  export let onPaymentSuccess: (request: any) => void;
+  export let onPaymentError: (error: string) => void;
+  const COUNTDOWN_INTERVAL = 30;
+
+  let amountInCrypto: number = 0;
+  let countdown: number = COUNTDOWN_INTERVAL;
+  let intervalId: NodeJS.Timeout;
+  let exchangeRate: number = 0;
+  let error: string = "";
+
+  $: isLoadingPrice = true;
+  $: isPaying = false;
+
+  const currencySymbol = selectedCurrency.symbol.includes(
+    selectedCurrency.network
+  )
+    ? selectedCurrency.symbol.split("-")[0]
+    : selectedCurrency.symbol;
+
+  async function fetchExchangeRate() {
+    if (isPaying) return;
+    try {
+      isLoadingPrice = true;
+      const response = await fetch(
+        `https://api.coinbase.com/v2/exchange-rates?currency=USD`
+      );
+      const data = await response.json();
+
+      let symbolCurrency = selectedCurrency.symbol.split("-")[0];
+      const symbolMap: { [key: string]: string } = {
+        fUSDC: "USDC",
+        fUSDT: "USDT",
+      };
+      const lookupSymbol = symbolMap[symbolCurrency] || symbolCurrency;
+      const rate = data.data.rates[lookupSymbol];
+      exchangeRate = parseFloat(rate);
+      amountInCrypto = amountInUSD * exchangeRate;
+    } catch (error) {
+      alert("Unable to fetch exchange rate. Please try again later");
+    } finally {
+      isLoadingPrice = false;
+    }
+  }
+
+  function startCountdown() {
+    countdown = COUNTDOWN_INTERVAL;
+    intervalId = setInterval(() => {
+      countdown--;
+      if (countdown === 0) {
+        fetchExchangeRate();
+        countdown = COUNTDOWN_INTERVAL;
+      }
+    }, 1000);
+  }
+
+  $: {
+    if (isPaying) {
+      clearInterval(intervalId);
+      countdown = 0;
+    } else if (countdown === 0) {
+      startCountdown();
+    }
+  }
+
+  onMount(() => {
+    fetchExchangeRate();
+    startCountdown();
+  });
+
+  onDestroy(() => {
+    clearInterval(intervalId);
+  });
+
+  function formatCryptoAmount(amount: number): string {
```

Comment:
Consider renaming to make it more clear what this function is doing
```suggestion
  function trimTrailingDecimalZeros(amount: number): string {
```